google-cloud-datastore/src/test/java/org/togglz/googleclouddatastore/repository/GoogleCloudDatastoreStateRepositoryTest.java

```diff
@@ -0,0 +1,306 @@
+        /*
+         * WHEN a feature without strategy is persisted
+         */
+        final FeatureState state = new FeatureState(TestFeature.F1).disable();
+        repository.setFeatureState(state);
+
+        /*
+         * THEN there should be a corresponding entry in the database
+         */
+        final Key key = datastore.newKeyFactory().setKind(repository.kind()).newKey(TestFeature.F1.name());
+        final Entity featureEntity = datastore.get(key);
+
+        assertEquals(false, featureEntity.getBoolean(GoogleCloudDatastoreStateRepository.ENABLED));
+        assertFalse(featureEntity.contains(GoogleCloudDatastoreStateRepository.STRATEGY_ID));
+        assertFalse(featureEntity.contains(GoogleCloudDatastoreStateRepository.STRATEGY_PARAMS_NAMES));
+        assertFalse(featureEntity.contains(GoogleCloudDatastoreStateRepository.STRATEGY_PARAMS_VALUES));
+
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testShouldSaveStateStrategyAndParameters() {
+
+        /*
+         * WHEN a feature without strategy is persisted
+         */
+        final FeatureState state = new FeatureState(TestFeature.F1)
+                .enable()
+                .setStrategyId("someId")
+                .setParameter("param", "foo");
+        repository.setFeatureState(state);
+
+        /*
+         * THEN there should be a corresponding entry in the database
+         */
+        final Key key = datastore.newKeyFactory().setKind(repository.kind()).newKey(TestFeature.F1.name());
+        final Entity featureEntity = datastore.get(key);
+
+        assertEquals(true, featureEntity.getBoolean(GoogleCloudDatastoreStateRepository.ENABLED));
+        assertEquals("someId", featureEntity.getString(GoogleCloudDatastoreStateRepository.STRATEGY_ID));
+        final StringValue param = StringValue.newBuilder("param").setExcludeFromIndexes(true).build();
+        assertThat(featureEntity.<StringValue>getList(GoogleCloudDatastoreStateRepository.STRATEGY_PARAMS_NAMES),
+                is(Arrays.asList(param)));
+        final StringValue foo = StringValue.newBuilder("foo").setExcludeFromIndexes(true).build();
+        assertThat(featureEntity.<StringValue>getList(GoogleCloudDatastoreStateRepository.STRATEGY_PARAMS_VALUES),
+                is(Arrays.asList(foo)));
+    }
+
+    @Test
+    public void shouldReturnNullWhenStateDoesntExist() {
+        /*
+         * GIVEN there is no feature state in the datastore WHEN the repository reads the state
+         */
+        final FeatureState state = repository.getFeatureState(TestFeature.F1);
+
+        /*
+         * THEN the properties should be set like expected
+         */
+        assertNull(state);
+    }
+
+    @Test
+    public void testShouldReadStateWithoutStrategyAndParameters() {
+
+        /*
+         * GIVEN a database row containing a simple feature state
+         */
+        update("F1", false, null, null, null);
```

Comment:
This method name and signature say nothing... 
Can you overload it?:
```
enable(String name, String strategyId, Map<String, String> params)
disable(String name, String strategyId, Map<String, String> params)
disable(String name, String strategyId)
disable(String name, String strategyId, Transaction txn)
```
You can also use ImmutableMap.of() instead null.